### ISSUE TITLE
caret and anchor stale after selectRange with listeners

### ISSUE THREAD

**Issue by Groostav** [FXMisc/RichTextFX#874]
If you attempt to use `selectRange` to update caret and anchor positions _that have attached listeners_, the caret and anchor potitions will be stale.

```kt
@Test fun `sscce`(){
    val underTest = StyleClassedTextArea()

    //comment out this line of code and the test passes
    val dontCare = Bindings.createStringBinding(Callable { "binded" }, underTest.caretPositionProperty(), underTest.anchorProperty())
    underTest.appendText("asdf")
    
    //act
    underTest.selectRange(1, 2)

    //assert
    assertThat(underTest.selectedText).isEqualTo("s") //OK
    assertThat(listOf(underTest.anchor, underTest.caretPosition)).isEqualTo(listOf(1, 2)) //fail
    //actual: 0, 2
}

```

**Comment by Groostav** [comment]
In my production code i have both the caret and anchor positions as 0, 0 when the position is set. I'll see if I can add another example here.

**Comment by Jugen** [comment]
I've had a look and it appears that this happens when a listener is added to the `anchorProperty` ?!
So it works correctly if we have:
```java
val dontCare = Bindings.createStringBinding
(
   () -> "binded", underTest.caretPositionProperty()
);
```
Will look further into this .....

**Comment by Groostav** [comment]
Thanks @Jugen , I can confirm this fixes the SSCCE and my problem in production. Any idea when this will make it into maven?

thanks again for the fix!

**Comment by Jugen** [comment]
I published to maven on Wed so hopefully today or tomorrow ....

**Comment by Groostav** [comment]
@Jugen Unfortunatly while this now makes it through our tests we discovered the problem is better but still bad in production.

Is it possible the same or a similar problem is happening to the anchor position?

**Comment by Jugen** [comment]
Umm, did you maybe mean caret position and not anchor position ?
Otherwise can you provide more information.

**Comment by Groostav** [comment]
@Jugen No I did mean anchor position. The caret appears to be working correctly. 

Firstly: the use case is to have a richTextFX text area, with a couple of labels bound to metadata about that text areas selection --a pretty common thing if you look at gedit or notepad++.

I've found that if I leave my selection-listeners on the text area it changes the behaviour of the text area. Text selection with the mouse becomes stunted and heavily prone to artificially moving the caret to the end of the document or the anchor to the beginning of the document. This is what lead to the original report. Since your fix the behaviour got better _but it is not corrected_. If I leave my selection-listeners on the text area, I still find it selecting the wrong ranges frequently.

I have tried to get this down to an SSCCE but have had difficulty.

But I have found a workaround: rather than bind to the `anchorProperty` and the `caretPositionProperty`, I'm simply binding to the `selectedTextProperty`.

I still need to do some work here, I'll let you know when I've got something reasonably actionable for you, but this one is tricky to run down.
